core/templates/dev/head/pages/library/Library.js

```diff
@@ -81,30 +81,35 @@ oppia.controller('Library', [
         $scope.libraryGroups = data.activity_summary_dicts_by_category;
 
         $scope.userIsLoggedIn = GLOBALS.userIsLoggedIn;
-        var activityDetails = {};
         if ($scope.userIsLoggedIn) {
           $scope.activitiesOwned = {explorations: {}, collections: {}};
           $http.get('/creatordashboardhandler/data')
             .then(function(response) {
               $scope.libraryGroups.forEach(function(activityGroups) {
-                var activity = activityGroups.activity_summary_dicts;
-
-                activity.forEach(function(activity) {
```

Comment:
In response to your earlier comment, I meant something like this (pseudocode):

```
    activitySummaryDicts.forEach(function(activity) {
      if activity.type is collection
        $scope.activitiesOwned.collections[activity.id] = false;
      else
        $scope.activitiesOwned.explorations[activity.id] = false;
    });

    response.data.explorations_list.forEach(function(ownedExploration) {
      $scope.activitiesOwned.explorations[ownedExploration].id = true;
    });
   
    response.data.explorations_list.forEach(function(ownedCollection) {
      $scope.activitiesOwned.collections[ownedCollection].id = true;
    });
```

That gets rid of the nested for loops. Would an approach like this not work?